### PR TITLE
NEW: Code cleanup (linting, raw queries).

### PR DESCRIPTION
**Pull request by mfendeksilverstripe**
# NEW: Code cleanup (raw queries).

* Raw queries replaced with ORM calls
* minor code cleanup

## Out of scope of this PR

* Deep integration

These will be covered separate change-sets

## Related issues

https://github.com/silverstripe/silverstripe-versioned-snapshots/issues/51

### CODE REVIEW

File: src/SnapshotPublishable.php [hunk 1]
```diff
@@ -544,31 +520,36 @@ public function getPreviousVersion($version = null): ?DataObject
 
     /**
      * @param array $snapShotIDs
-     * @return SQLSelect
+     * @return DataList
+     * @throws Exception
      */
-    protected function publishableItemsQuery(array $snapShotIDs): SQLSelect
+    protected function publishableItemsList(array $snapShotIDs): DataList
```
Comment (blueo) [inline src/SnapshotPublishable.php:526]: suggest: `publishableItems`

File: src/SnapshotPublishable.php [hunk 2]
```diff
@@ -544,31 +520,36 @@ public function getPreviousVersion($version = null): ?DataObject
...
+        $singleton = SnapshotItem::singleton();
+        $itemTable = DataObject::getSchema()->tableName($singleton->ClassName);
+
+        return SnapshotItem::get()
+            ->filter([
+                'Snapshot.ID' => $snapShotIDs,
+                'WasPublished' => 0,
+                'WasDeleted' => 0,
+            ])
+            ->applyRelation('Snapshot.OriginHash', $snapshotOriginHashColumn)
```
Comment (blueo) [inline src/SnapshotPublishable.php:537]: this function can be a bit confusing - suggest passing a variable in rather that just having it created in the arguments

File: src/SnapshotPublishable.php [hunk 3]
```diff
@@ -723,30 +704,40 @@ public function getIntermediaryObjects(): array
...
+            return ArrayList::create()->setDataClass($singleton->ClassName);
+        }
+
+        $itemTable = DataObject::getSchema()->tableName($singleton->ClassName);
 
         return SnapshotItem::get()
-            ->innerJoin($snapshotTable, "\"$snapshotTable\".\"ID\" = \"$itemTable\".\"SnapshotID\"")
-            ->leftJoin($itemTable, "\"ChildItem\".\"ParentID\" = \"$itemTable\".\"ID\"", "ChildItem")
+            ->filter([
+                // Intentionally forcing a join here
```
Comment (blueo) [inline src/SnapshotPublishable.php:723]: not sure what this is referring to in the filter clause?

File: src/SnapshotPublishable.php [hunk 4]
```diff
@@ -723,30 +704,40 @@ public function getIntermediaryObjects(): array
...
                 // Only get the items that were the subject of a user's action
-                "(
-                    \"$snapshotTable\" . \"OriginHash\" = \"$itemTable\".\"ObjectHash\" AND
-                    \"ChildItem\".\"ID\" IS NULL
-                ) OR (
-                    \"$snapshotTable\" . \"OriginHash\" != \"$itemTable\".\"ObjectHash\" AND
-                    \"$itemTable\".\"ParentID\" != 0
-                )"
+                sprintf(
+                    '(%1$s = "%3$s"."ObjectHash" AND %2$s IS NULL) OR '
```
Comment (blueo) [inline src/SnapshotPublishable.php:731]: [nitpick] this is quite difficult to read though it is efficient. Might be clearer if it was split into a few parts - but this is just preference

**Review by blueo — CHANGES_REQUESTED** [review]
Couldn't see any major logical changes - mainly made suggestions about naming and clarity :)

File: src/SnapshotPublishable.php (hunk 2 above)
Comment (mfendeksilverstripe) [reply src/SnapshotPublishable.php:537]: Can you elaborate on this one? I'm using this as suggested in the documentation of this function. Note that the variable is passed by reference, it's essentially a return value as we need it later.

Comment (blueo) [reply src/SnapshotPublishable.php:537]: yeah so the variable is declared in the argument which can be a bit surprising - its technically correct but I just did a double take trying to figure out where the variable was declared so thought could be clearer to do something like:

```
$snapshotOriginHashColumn = null;
$list->applyRelation('Snapshot.OriginHash', $snapshotOriginHashColumn);
```

its probably more of a nitpick as it works but it a little unusual

File: src/SnapshotPublishable.php (hunk 3 above)
Comment (mfendeksilverstripe) [reply src/SnapshotPublishable.php:723]: added comment for more clarity

File: src/SnapshotPublishable.php (hunk 4 above)
Comment (mfendeksilverstripe) [reply src/SnapshotPublishable.php:731]: Added comment for clarity.

**Comment by mfendeksilverstripe** [comment]
@blueo I've pushed up new changes, please review.

**Review by blueo — APPROVED** [review]
love your work @mfendeksilverstripe - in general I find the the ORM is vastly easier to read and understand so this feels like a great update.

**Pull request merged**